src/uproot/source/http.py

```diff
@@ -298,25 +299,24 @@ def multifuture(source, ranges, futures, results):
         )
 
         def task(resource):
+            nonlocal connection
```

Comment:
That's right: `connection[0]` was a Python2ism. `global` is not the right scope, and `nonlocal` didn't exist.

Reply:
Hmm, you're right. At one point I was making this an implicit thing, but there's sense in exposing it.